tests/integration/states/test_vmc_security_groups.py

```diff
@@ -0,0 +1,246 @@
+        name="present",
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        domain_id=domain_id,
+        security_group_id=security_group_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+        display_name=updated_display_name,
+    )
+    response_json = response.json
+    result = list(response_json.values())[0]
+    changes = result["changes"]
+
+    assert changes["old"]["display_name"] != changes["new"]["display_name"]
+    assert changes["new"]["display_name"] == updated_display_name
+    assert result["comment"] == "Updated Security group {}".format(security_group_id)
+
+    # Invoke present state to update security group with tags field
+    response = salt_call_cli.run(
+        "state.single",
+        "vmc_security_groups.present",
+        name="present",
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        domain_id=domain_id,
+        security_group_id=security_group_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+        tags=updated_tags,
+    )
+    response_json = response.json
+    result = list(response_json.values())[0]
+    changes = result["changes"]
+
+    assert changes["new"]["tags"] == updated_tags
+    assert result["comment"] == "Updated Security group {}".format(security_group_id)
+
+    # Invoke absent to delete the security group
+    response = salt_call_cli.run(
+        "state.single",
+        "vmc_security_groups.absent",
+        name="absent",
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        domain_id=domain_id,
+        security_group_id=security_group_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+    response_json = response.json
+    result = list(response_json.values())[0]
+    changes = result["changes"]
+
+    assert changes["new"] is None
+    assert changes["old"]["id"] == security_group_id
+    assert result["comment"] == "Deleted Security group {}".format(security_group_id)
+
+    # Invoke absent when security group is not present
+    response = salt_call_cli.run(
+        "state.single",
+        "vmc_security_groups.absent",
+        name="absent",
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        domain_id=domain_id,
+        security_group_id=security_group_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+    response_json = response.json
+    result = list(response_json.values())[0]
+    changes = result["changes"]
+    # assert no changes are done
+    assert changes == {}
+    assert result["comment"] == "No Security group found with Id {}".format(security_group_id)
```

Comment:
**praise** this style of integration test for CRUD operations is probably much better.